content/en/docs/develop/porting.md

```diff
@@ -614,23 +614,338 @@ To make a patch:
+
+1. **Config.uk**:
+   A [Kconfig-like](https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html) snippet used to populate Unikraft's menu with application-specific options.
+
+1. **exportsyms.uk**:
+   A text file where each line contains the name of one symbol that should be exported to other libraries.
+   This file usually contains only the `main()` function for an application that is developed/ported as a single library to Unikraft.
+   If an `exportsyms.uk` file is not present, all symbols will be exported.
+
+1. **extra.ld**:
+   [Optional] Contains an amendment to the main linker script.
+
+In the tutorial below, we will focus on files belonging to an application (i.e. not to a library).
+The process of porting a library is very similar, just change the `APP` prefix of the variable names to `LIB`.
+
+#### Makefile
+
+The `Makefile` is generally short and simple and might remind you of Linux kernel modules that are built off-tree.
+For most applications the `Makefile` should contain no more than the following:
+
+```Makefile
+UK_ROOT  ?= $(PWD)/../../unikraft
+UK_LIBS  ?= $(PWD)/../../libs
+UK_PLATS ?= $(PWD)/../../plats
+LIBS  := $(UK_LIBS)/lib1:$(UK_LIBS)/lib2:$(UK_LIBS)/libN
+PLATS ?=
+
+all:
+        @make -C $(UK_ROOT) A=$(PWD) L=$(LIBS) P=$(PLATS)
+
+$(MAKECMDGOALS):
+        @make -C $(UK_ROOT) A=$(PWD) L=$(LIBS) P=$(PLATS) $(MAKECMDGOALS)
+```
+
+You can notice that the `Makefile` is just a wrapper around the [Unikraft core `Makefile`](https://github.com/unikraft/unikraft/blob/staging/Makefile).
+You can get the same result if you use:
+
+```console
+$ make -C ../../unikraft A=. L=../../libs/lib1:../../libs/lib2:../../libs/libN
+```
+
+#### Makefile.uk
+
+The Unikraft build system provides a number of functions and macros to make it easier to write the `Makefile.uk` file.
+Also, ensure that all variables that you define begin with `APP[NAME]_` or `LIB[NAME]_` (e.g., `APPHELLOWORLD_`) so that they are properly namespaced.
+
+The first thing to do is to call the Unikraft `addlib` function to register the application with the system (note the letters "lib": everything in Unikraft is ultimately a library).
+This function call will also populate `APPNAME_BASE` (containing the directory path to the application sources) and `APPNAME_BUILD` (containing the directory path to the application's build directory):
+
+```Makefile
+$(eval $(call addlib,appname))
+```
+
+where `name` would be replaced by your application's name.
+In case your main application code should be downloaded as an archive from a remote server, the next step is to set up a variable to point to a URL with the application sources (or objects, or pre-linked libraries, see further below), and to call Unikraft's `fetch` command to download and unpack those sources (`APPNAME_ORIGIN` is populated containing the directory path to the extracted files):
```

Comment:
```suggestion
In case your main application code should be downloaded as an archive from a remote server, the next step is to set up a variable to point to a URL with the application sources (or objects, or pre-linked libraries, see further below), and to call [the Unikraft `fetch` function](https://github.com/unikraft/unikraft/blob/staging/support/build/Makefile.rules#L456) to download and unpack those sources (`APPNAME_ORIGIN` is populated containing the directory path to the extracted files):
```